src/App.tsx

```diff
@@ -9,36 +9,77 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { client } from './utils/fetchClient';
+import { Post } from './types/Post';
+
 export const App: React.FC = () => {
+  const [isUserSelected, setIsUserSelected] = useState(false);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [isNoPosts, setIsNoPosts] = useState(false);
```

Comment:
Do you need this? You can just check the length

Reply:
you are right, fixed